File: src/main/scala/resonantinduction/core/prefab/node/TilePressureNode.java
package resonantinduction.core.prefab.node;

import net.minecraft.block.material.Material;
import net.minecraftforge.common.util.ForgeDirection;

/**
 * Created by robert on 8/15/2014.
 */
public class TilePressureNode extends TileTankNode
{
	public TilePressureNode(Material material)
	{
		super(material);
		tankNode_$eq(new NodePressure(this));
	}

	public NodePressure getPressureNode()
	{
		return (NodePressure) tankNode();
	}

	public int getPressure(ForgeDirection direction)
	{
		return getPressureNode().getPressure(direction);
	}
}

File: src/main/scala/resonantinduction/core/prefab/node/TilePressureNode.scala
package resonantinduction.core.prefab.node

import net.minecraft.block.material.Material
import net.minecraftforge.common.util.ForgeDirection
import resonant.lib.prefab.fluid.NodeFluid

class TilePressureNode(material: Material) extends TileFluidProvider(material: Material)
{
  //Constructor
  fluidNode == new NodePressure(this)

  override protected var fluidNode: NodeFluid = new NodePressure(this)

  def getPressureNode: NodePressure =
  {
    return fluidNode.asInstanceOf[NodePressure]
  }

  def getPressure(direction: ForgeDirection): Int =
  {
    return getPressureNode.pressure(direction)
  }
}